components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +264,136 @@ func NewAdxl345(
 			}
 		}
 	})
+	sensor.interruptsFound = make(map[InterruptID]int)
+	if err := sensor.readInterrupts(sensor.cancelContext); err != nil {
+		// shut down goroutine reading sensor in the background
+		sensor.cancelFunc()
+		return nil, err
+	}
+	if err := sensor.configureInterruptRegisters(ctx, interruptConfigurations[IntMapAddr]); err != nil {
+		// shut down goroutine reading sensor in the background
+		sensor.cancelFunc()
+		return nil, err
+	}
+
+	interruptMap := map[string]board.DigitalInterrupt{}
+	if (cfg.SingleTap != nil) && (cfg.SingleTap.InterruptPin != "") {
+		interruptMap, err = addInterruptPin(b, cfg.SingleTap.InterruptPin, interruptMap)
+		if err != nil {
+			// shut down goroutine reading sensor in the background
+			sensor.cancelFunc()
+			return nil, err
+		}
+	}
+
+	if (cfg.FreeFall != nil) && (cfg.FreeFall.InterruptPin != "") {
+		interruptMap, err = addInterruptPin(b, cfg.FreeFall.InterruptPin, interruptMap)
+		if err != nil {
+			// shut down goroutine reading sensor in the background
+			sensor.cancelFunc()
+			return nil, err
+		}
+	}
+
+	for _, interrupt := range interruptMap {
+		ticksChan := make(chan board.Tick)
+		sensor.startInterruptMonitoring(interrupt, ticksChan)
+	}
 
 	return sensor, nil
 }
 
+func (adxl *adxl345) startInterruptMonitoring(interrupt board.DigitalInterrupt, ticksChan chan board.Tick) {
+	utils.PanicCapturingGo(func() {
+		interrupt.AddCallback(ticksChan)
+		defer interrupt.RemoveCallback(ticksChan)
+		for {
+			select {
+			case <-adxl.cancelContext.Done():
+				return
+			case tick := <-ticksChan:
+				if tick.High {
+					func() {
+						utils.UncheckedError(adxl.readInterrupts(adxl.cancelContext))
+					}()
```

Comment:
I'd remove the `func() {...}()` from the outside, and just call `utils.UncheckedError` directly.